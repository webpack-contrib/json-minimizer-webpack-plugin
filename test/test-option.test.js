const JsonMinimizerPlugin = require("../src/index");

const {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} = require("./helpers");

describe('when applied with "test" option', () => {
  let compiler;

  beforeEach(() => {
    jest.clearAllMocks();

    const testJsonId = "./test/foo-[0-4].json";

    compiler = getCompiler(testJsonId);
  });

  it("matches snapshot with empty value", async () => {
    new JsonMinimizerPlugin().apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('matches snapshot for a single "test" value (RegExp)', async () => {
    new JsonMinimizerPlugin({
      test: /foo-[1-3]\.json/,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('matches snapshot for multiple "test" value (RegExp)', async () => {
    new JsonMinimizerPlugin({
      test: [/foo-[0]\.json/, /foo-[1-2]\.json/],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
