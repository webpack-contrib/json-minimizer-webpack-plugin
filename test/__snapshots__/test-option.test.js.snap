// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with "test" option matches snapshot for a single "test" value (RegExp): assets 1`] = `
{
  "test/foo-0.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "test/foo-1.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-2.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-3.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-4.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
}
`;

exports[`when applied with "test" option matches snapshot for a single "test" value (RegExp): errors 1`] = `[]`;

exports[`when applied with "test" option matches snapshot for a single "test" value (RegExp): warnings 1`] = `[]`;

exports[`when applied with "test" option matches snapshot for multiple "test" value (RegExp): assets 1`] = `
{
  "test/foo-0.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-1.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-2.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-3.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "test/foo-4.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
}
`;

exports[`when applied with "test" option matches snapshot for multiple "test" value (RegExp): errors 1`] = `[]`;

exports[`when applied with "test" option matches snapshot for multiple "test" value (RegExp): warnings 1`] = `[]`;

exports[`when applied with "test" option matches snapshot with empty value: assets 1`] = `
{
  "test/foo-0.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-1.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-2.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-3.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "test/foo-4.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
}
`;

exports[`when applied with "test" option matches snapshot with empty value: errors 1`] = `[]`;

exports[`when applied with "test" option matches snapshot with empty value: warnings 1`] = `[]`;
